pkg/trace/api/debugger.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/fargate"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// logsIntakeURLTemplate specifies the template for obtaining the intake URL along with the site.
+	logsIntakeURLTemplate = "https://http-intake.logs.%s/v1/input"
+)
+
+// debuggerProxyHandler returns an http.Handler proxying requests to the logs intake. If the logs intake url cannot be
+// parsed, the returned handler will always return http.StatusInternalServerError with a clarifying message.
+func (r *HTTPReceiver) debuggerProxyHandler() http.Handler {
+	tags := fmt.Sprintf("host:%s,default_env:%s,agent_version:%s", r.conf.Hostname, r.conf.DefaultEnv, info.Version)
+	if orch := r.conf.FargateOrchestrator; orch != fargate.Unknown {
+		tags = tags + ",orchestrator:fargate_" + strings.ToLower(string(orch))
+	}
+	intake := fmt.Sprintf(logsIntakeURLTemplate, config.DefaultSite)
+	if v := config.Datadog.GetString("apm_config.debugger_dd_url"); v != "" {
+		intake = v
+	} else if site := config.Datadog.GetString("site"); site != "" {
+		intake = fmt.Sprintf(logsIntakeURLTemplate, site)
+	}
+	target, err := url.Parse(intake)
+	if err != nil {
+		log.Criticalf("Error parsing snapshot intake URL %q: %v", intake, err)
```

Comment:
```suggestion
		log.Criticalf("Error parsing live debugger intake URL %q: %v", intake, err)
```
"snapshot" is too ambiguous I think, considering that the trace-agent now handles multiple products...

The below error (in the error handler) is fine, that's within context.